core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -393,11 +393,13 @@ public ClientOptions options() {
 
     @Override
     public EndpointGroup endpointGroup() {
+        checkState(endpointGroup != null, "Should call init(endpoint) before invoking this method.");
```

Comment:
How about using `assert`? It's perhaps a bug if this method is exposed to a user before `init()` is invoked.